File: HotDesk_PRAI/Library/Library.Root/TemplateField/historyfield.vb
Imports System.Web.UI.WebControls

Public Class historyfield
    Implements System.Web.UI.ITemplate

    Public Const LabelHeaderID As String = "lblhis"
    Public Const LiteralItemID As String = "ltrhisitem"
    Private plittype As ListItemType

    Public Sub New(ByVal type As ListItemType)
        MyBase.New()
        Me.plittype = type
    End Sub

    Public Sub InstantiateIn(ByVal container As System.Web.UI.Control) Implements System.Web.UI.ITemplate.InstantiateIn
        Dim ltritem As Literal
        Dim lbdel As Label

        Select Case Me.plittype
            Case listItemType.Header
                lbdel = New Label()
                lbdel.Text = "Audit Trail"
                lbdel.ID = "lblhis"
                lbdel.EnableViewState = False
                container.Controls.Add(lbdel)
            Case listItemType.Item
                ltritem = New Literal()
                ltritem.ID = "ltrhisitem"
                ltritem.EnableViewState = False
                container.Controls.Add(ltritem)
        End Select
    End Sub
End Class

File: TMS_HotDesk/Library/Library.Root/TemplateField/historyfield.cs
using System.Web.UI.WebControls;

public class historyfield : System.Web.UI.ITemplate
{
    public const string LabelHeaderID = "lblhis";
    public const string LiteralItemID = "ltrhisitem";
    private ListItemType plittype;

    public historyfield(ListItemType type)
    {
        this.plittype = type;
    }

    public void InstantiateIn(System.Web.UI.Control container)
    {
        Literal ltritem;
        Label lbdel;

        switch (this.plittype)
        {
            case ListItemType.Header:
                lbdel = new Label();
                lbdel.Text = "Audit Trail";
                lbdel.ID = "lblhis";
                lbdel.EnableViewState = false;
                container.Controls.Add(lbdel);
                break;
            case ListItemType.Item:
                ltritem = new Literal();
                ltritem.ID = "ltrhisitem";
                ltritem.EnableViewState = false;
                container.Controls.Add(ltritem);
                break;
        }
    }
}
